src/days-2-4-adv/adv.py

```diff
@@ -39,13 +42,89 @@
 
 # Make a new player object that is currently in the 'outside' room.
 
+# player = Player(room['outside'])
+# print(player.currentRoom.description)
+
+p = Player(input('\n\n\n\n\nWhat is your name?'), room['outside'])
+
 # Write a loop that:
 #
 # * Prints the current room name
 # * Prints the current description (the textwrap module might be useful here).
 # * Waits for user input and decides what to do.
-#
 # If the user enters a cardinal direction, attempt to move to the room there.
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+while True:
+    time.sleep(1)
+    cmd = input('\n\n\n\n\nYou are in the main menu. Are you ready? \n=>')
+
+    if cmd.upper() == 'YES':
+        while True:
+            currentRoom = p.currentRoom
+            def error():
+                time.sleep(2)
+                print('\nYou cannot go in that direction! Try another direction!')
+            time.sleep(2)
+            print('\n\n\nYour location is: ') 
+            time.sleep(1)
+            print(currentRoom)
+            time.sleep(1)
+            cmd = input('\n\n\nWhere do you want to go?\n=>')
+
+            if cmd.upper() == 'N':
```

Comment:
You might want to consider combining all your directional commands into one if statement.  You could do something like `elif cmd in ["n", "s", "e", "w"]:` then nest a second if to check if the `hasattr(player.currentRoom, ''.join([cmd + '_to']))` then use getattr to change the room.
https://docs.python.org/3/library/functions.html#getattr

Reply:
I refactored that part of the code according to the guided demo solution. Thanks! @wvandolah